htdocs/xoops_lib/Xoops.php

```diff
@@ -733,7 +733,7 @@ public function isAdmin()
      *
      * @param boolean $optional true if failure to load handler should be considered a warning, not an error
      *
-     * @return \Xoops\Core\Kernel\Handlers\XoopsBlockHandler
+     * @return \Xoops\Core\Kernel\Handlers\XoopsBlockHandler|\Xoops\Core\Kernel\XoopsPersistableObjectHandler
```

Comment:
These were made specific to the expected handler so that some other tools would quit complaining. The returns from these methods will be a specific subclass of XoopsPersitableObjectHandler, which will be expected by the caller. If we declare to return the base class as well, that will trigger other warnings is tools like scrutinizer.

I'm not convinced this fixes anything, it just moves it around. The handler factory is not really complete, so perhaps we should address this there instead?